provider/kubernetes/kubernetes_test.go

```diff
@@ -2262,6 +2263,297 @@ func TestBasicAuthInTemplate(t *testing.T) {
+	testIngress := v1beta1.Ingress{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test-ingress",
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "ep1.example.com",
+				},
+				{
+					Host: "ep2.example.com",
+				},
+			},
+			TLS: []v1beta1.IngressTLS{{SecretName: "test-secret"}},
+		},
+	}
+
+	tests := []struct {
+		desc      string
+		ingress   *v1beta1.Ingress
+		client    Client
+		result    []*tls.Configuration
+		errResult string
+	}{
+		{
+			desc:    "error_api_client",
+			ingress: &testIngress,
+			client: clientMock{
+				apiSecretError: errors.New("api secret error"),
+			},
+			errResult: "secret testing/test-secret does not exist: api secret error",
+		},
+		{
+			desc:      "secret_not_found",
+			ingress:   &testIngress,
+			client:    clientMock{},
+			errResult: "secret testing/test-secret does not exist",
+		},
+		{
+			desc:    "secret_crt_missing",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.key": []byte("tls-key"),
+						},
+					},
+				},
+			},
+			errResult: "secret testing/test-secret must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.crt'",
+		},
+		{
+			desc:    "secret_key_missing",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.crt": []byte("tls-crt"),
+						},
+					},
+				},
+			},
+			errResult: "secret testing/test-secret must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.key'",
+		},
+		{
+			desc:    "default_wildcard_behavior",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.crt": []byte("tls-crt"),
+							"tls.key": []byte("tls-key"),
+						},
+					},
+				},
+			},
+			result: []*tls.Configuration{
+				{
+					Certificate: &tls.Certificate{
+						CertFile: tls.FileOrContent("tls-crt"),
+						KeyFile:  tls.FileOrContent("tls-key"),
+					},
+					EntryPoints: []string{
+						"ep1.example.com",
+						"ep2.example.com",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			tlsConfigs, err := getTLSConfigurations(test.ingress, test.client)
+
+			if test.errResult == "" {
+				assert.Nil(t, err)
+			} else {
+				assert.EqualError(t, err, test.errResult)
+			}
+
+			if !reflect.DeepEqual(tlsConfigs, test.result) {
+				t.Errorf("Result should be %+v but was %+v", test.result, tlsConfigs)
+			}
```

Comment:
I don't think we need to give any guarantees for `tlsConfigs` when we have a non-nil error. Consequently, I'd only validate the configuration during the happy path and modify the test accordingly; e.g.:

```go
if test.errResult != "" {
    assert.EqualError(t, err, test.errResult)
} else {
    assert.Nil(t, err)
    assert.Equal(t, tlsConfigs, test.result)
}
```